internal/storage/v2/elasticsearch/tracestore/to_dbmodel.go

```diff
@@ -48,31 +54,64 @@ func ToDBModel(td ptrace.Traces) []dbmodel.Span {
 	return batches
 }
 
-func resourceSpansToDbSpans(resourceSpans ptrace.ResourceSpans) []dbmodel.Span {
+type toDBModel struct {
+	allTagsAsFields   bool
+	tagKeysAsFields   map[string]bool
+	tagDotReplacement string
+}
+
+func newToDBModel(allTagsAsFields bool, tagKeysAsFields map[string]bool, tagDotReplacement string) *toDBModel {
+	return &toDBModel{
+		allTagsAsFields:   allTagsAsFields,
+		tagKeysAsFields:   tagKeysAsFields,
+		tagDotReplacement: tagDotReplacement,
+	}
+}
+
+func (t *toDBModel) embedTagDotReplacement(keyValues []dbmodel.KeyValue) ([]dbmodel.KeyValue, map[string]any) {
```

Comment:
what is the primary responsibility of this function? Try to write a sentence (in comment) and then choose a concise name that reflects that description.